containers/daap-create-metadata/src/var/task/create_metadata.py

```diff
@@ -1,74 +1,112 @@
             "image_version": os.getenv("VERSION", "unknown"),
             "base_image_version": os.getenv("BASE_VERSION", "unknown"),
         }
     )
 
+    def format_response(response_code: int, event: dict, body_dict: dict) -> dict:
+        """
+        Generate a response to return to API Gateway that contains the initial event,
+        response code, and contents of the response body
+        (i.e. body['message'] or body['error']['message'])
+        """
+        response_body = {"input": event}
+        response_body.update(body_dict)
+        response = {
+            "statusCode": response_code,
+            "headers": {"Content-Type": "application/json"},
+            "body": json.dumps(response_body),
+        }
+        return response
+
     logger.info(f"event: {event}")
+    response_code = HTTPStatus.SUCCESS.value
+    response_body = {}
+    data_product_name = None
 
-    request_body = json.loads(event["body"])
+    http_method = event.get("httpMethod")
+    event_body = json.loads(event.get("body"))
 
     try:
-        data_product_name = request_body["metadata"]["name"]
+        data_product_name = event_body["metadata"]["name"]
     except KeyError:
-        logger.error("The name of the data product is missing from the metadata.")
-        return {
-            "statusCode": 400,
-            "headers": {"Content-Type": "application/json"},
-            "body": json.dumps(
-                {
-                    "message": "Data product name is missing, it must be specified in the metadata against the 'name' key"  # noqa E501
-                }
-            ),
-        }
+        response_code = HTTPStatus.BAD_REQUEST.value
+        error = "Data product name is missing, it must be specified in the metadata against the 'name' key"  # noqa E501
+        response_body = {"error": {"message": error}}
+        logger.error(error)
+        return format_response(response_code, event, response_body)
 
     logger.add_extras({"data_product_name": data_product_name})
 
+    if http_method == "POST":
+        pass
+    else:
+        error = f"Sorry, {http_method} isn't allowed."
+        response_body = {"error": {"message": error}}
+        response_code = HTTPStatus.METHOD_NOT_ALLOWED.value
+        logger.error(error)
+        return format_response(response_code, event, response_body)
+
     data_product_metadata = DataProductMetadata(data_product_name, logger)
 
     if not data_product_metadata.metadata_exists:
-        data_product_metadata.validate(request_body["metadata"])
-    else:
-        return {
-            "statusCode": 400,
-            "headers": {"Content-Type": "application/json"},
-            "body": json.dumps(
-                {
-                    "data_product_name": data_product_name,
-                    "message": "Your data product already has a version 1 registered metadata.",
-                }
-            ),
-        }
+        data_product_metadata.validate(event_body["metadata"])
```

Comment:
It may be a style choice, but I favour preserving the consistency of 

`try/if` _happy path_ 
`except/else` _unhappy path_

throughout the handler